flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/resource/NodeResourceUtil.java

```diff
@@ -34,8 +34,9 @@
 	 * Build resourceSpec from managedMem.
 	 */
 	public static ResourceSpec fromManagedMem(int managedMem) {
-		ResourceSpec.Builder builder = ResourceSpec.newBuilder();
-		builder.setManagedMemoryInMB(managedMem);
+		ResourceSpec.Builder builder = ResourceSpec.newBuilder(0.0, 0);
+		// TODO: before operators separate on-heap/off-heap managed memory, we use on-heap managed memory to denote total managed memory
+		builder.setOnHeapManagedMemoryMB(managedMem);
```

Comment:
This is an util class for table planner only. It's in the module `flink-table-planner-blink`. It's not used for DataSet API.

We do not plan to enable fine grained resource management for DataSet API. It will use `UNKNOWN` resource profiles/specs only. For release 1.10, the plan is to enable fine grained resource management for blink table/sql jobs only. So there shouldn't be a problem.

Reply:
Offline discussion: defining and matching the managed memory requirement with arbitrary type can be a follow-up.